### PR TITLE
evy: Add `pi` as global constant to Evy

### PR DESCRIPTION
**Pull request by juliaogris**
Add `pi` as global variable to Evy. Pi is often needed
when calculating radians from degrees and/or placing shapes in a
circle.

In all examples where we defined `pi` as a local constant, we now use
the pre-defined global one.

### CODE REVIEW

File: frontend/lab/samples/ifs/pi.md [hunk 1]
```diff
@@ -124,7 +124,6 @@ And at the bottom of the loop, add:
 ```evy
 while // ...
   // ...
-  pi := reds / i * 4
```
Comment (camh-) [inline frontend/lab/samples/ifs/pi.md:127]: This is estimating π, isn't it? So you should not be removing this I think. Perhaps this sample could show the delta between the builtin π and the estimated π?

**Review by camh- — COMMENTED** [review]
A couple of thoughts outside individual lines:

The PR/commit message says "constant" but there are no constants in Evy. It really is a variable that could be re-assigned if wanted.

π should be added to the builtins docs somewhere.

File: frontend/lab/samples/ifs/pi.md (hunk 1 above)
Comment (juliaogris) [reply frontend/lab/samples/ifs/pi.md:127]: that's a great idea!

Comment (juliaogris) [reply frontend/lab/samples/ifs/pi.md:127]: fixed.

File: pkg/evaluator/builtin.go [hunk 2]
```diff
@@ -138,9 +147,19 @@ func newBuiltins(rt Runtime) builtins {
 		"input":   {Name: "input", Params: inputParams},
 		"animate": {Name: "animate", Params: numParam},
 	}
-	globals := map[string]*parser.Var{
-		"err":    {Name: "err", T: parser.BOOL_TYPE},
-		"errmsg": {Name: "errmsg", T: parser.STRING_TYPE},
+	globals := map[string]global{
+		"err": {
+			val:       &boolVal{V: false},
+			parserVar: &parser.Var{Name: "err", T: parser.BOOL_TYPE},
```
Comment (camh-) [inline pkg/evaluator/builtin.go:153]: nit: can you swap these two over (and below). Then it matches the order declared in the struct but also doesn't seem backwards to me: var then value is the ordering I'd expect.

**Review by camh- — APPROVED** [review]
:green_apple: with comment

File: pkg/evaluator/builtin.go (hunk 2 above)
Comment (juliaogris) [reply pkg/evaluator/builtin.go]: of course, agreed. fixed.

**Pull request merged**